File: server/src/main/java/com/khanabook/saas/sync/dto/PushSyncResponse.java
package com.khanabook.saas.sync.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PushSyncResponse {
	private List<Long> successfulLocalIds;
	private List<Long> failedLocalIds;
	private Map<Long, Long> localToServerIdMap;

	public PushSyncResponse(List<Long> successfulLocalIds, List<Long> failedLocalIds) {
		this.successfulLocalIds = successfulLocalIds;
		this.failedLocalIds = failedLocalIds;
		this.localToServerIdMap = new java.util.HashMap<>();
	}
}

File: Android/app/src/main/java/com/khanabook/lite/pos/data/remote/dto/PushSyncResponse.kt
package com.khanabook.lite.pos.data.remote.dto

import com.google.gson.annotations.SerializedName


data class PushSyncResponse(
    @SerializedName("successfulLocalIds")
    val successfulLocalIds: List<Long>,
    @SerializedName("failedLocalIds")
    val failedLocalIds: List<Long>,
    @SerializedName("localToServerIdMap")
    val localToServerIdMap: Map<Long, Long>? = null,
    @SerializedName("failedReasons")
    val failedReasons: Map<Long, String>? = null
)
